projects/cli-files/go-cat/cmd/root.go

```diff
@@ -0,0 +1,59 @@
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+)
+
+func Execute() {
+
+	// if there is no argument given then error
+	if len(os.Args) < 2 {
+		fmt.Fprintf(os.Stderr, "no filename provided\n")
+		os.Exit(2)
+	}
+
+	// if there is an argument provided then save that argument as the filename
+	filename := os.Args[1]
+
+	// print out the filename to check
+	// fmt.Println("--- DEBUG filename", filename)
+
+	// check if the file exists
+	// fileInfo, err := os.Stat(filename)
+	_, err := os.Stat(filename)
+
+	// if the file does not exist then error
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "no such file or directory: %s\n", filename)
```

Comment:
what if it's a directory? what's the output of `cat` for directories?

Reply:
I didn't know the answer to this, so I ran it:
```
$ cat go-ls
cat: go-ls: Is a directory
```
Ok so it should say `"%s: Is a directory"` if it's a directory, which I will have to implement another check for

Reply:
Implemented